### ISSUE TITLE
`minimal_primes` fails where it shouldn't 

### ISSUE THREAD

**Issue by HechtiDerLachs** [oscar-system/Oscar.jl#3773]
The following ideal is passed to Singular after expansion of the coefficient field of the principal ideal 
generated by [this irreducible polynomial](https://github.com/oscar-system/Oscar.jl/files/15401577/f.txt) (already posted on #3711) .

The expectation is that the computation should finish in short time. 
```julia
R, (a, b, x, y, z) = QQ[:a, :b, :x, :y, :z]
g = QQMPolyRingElem[-x^3 + x*z^8 - x + y^2, a^8 + 1, -a^6 + a^4 - a^2 + b^2]
J = ideal(R, g)
minimal_primes(J)
```
@afkafkafk13 observed that the order of the variables probably plays a decisive role. Hence I include another version here for convenience. It also does not finish for me. 
```julia
R, (x, y, z, a, b) = QQ[:x, :y, :z, :a, :b]
g = QQMPolyRingElem[-x^3 + x*z^8 - x + y^2, a^8 + 1, -a^6 + a^4 - a^2 + b^2]
J = ideal(R, g)
minimal_primes(J)
```
For me the stacktrace when aborting is
```
julia> minimal_primes(J)
^CERROR: InterruptException:
Stacktrace:
 [1] call_singular_library_procedure
   @ ~/.julia/packages/CxxWrap/5IZvn/src/CxxWrap.jl:624 [inlined]
 [2] low_level_caller_rng(lib::String, name::String, ring::Singular.PolyRing{…}, args::Singular.sideal{…})
   @ Singular ~/.julia/packages/Singular/C3EMh/src/caller.jl:379
 [3] minAssGTZ(ring::Singular.PolyRing{Singular.n_Q}, args::Singular.sideal{Singular.spoly{Singular.n_Q}})
   @ Singular.LibPrimdec ~/.julia/packages/Singular/C3EMh/src/Meta.jl:45
 [4] minimal_primes(I::MPolyIdeal{QQMPolyRingElem}; algorithm::Symbol, cache::Bool)
   @ Oscar ~/Kaiserslautern/neuer_Oscar_Klon/Oscar.jl/src/Rings/mpoly-ideals.jl:1079
 [5] minimal_primes(I::MPolyIdeal{QQMPolyRingElem})
   @ Oscar ~/Kaiserslautern/neuer_Oscar_Klon/Oscar.jl/src/Rings/mpoly-ideals.jl:1067
 [6] top-level scope
   @ REPL[28]:1
Some type information was truncated. Use `show(err)` to see complete types.
```
I am wondering whether this suffers from similar problems as discussed on #3711. But @fingolfin already remarked that this might not be the case.

**Comment by joschmitt** [comment]
Regarding the possible connection to #3711: I just tried these with Singular.jl 0.23.2 and they didn't finish after 3 minutes.

**Comment by HechtiDerLachs** [comment]
Yes, similar observations were made by @wdecker . Thanks for trying on your machine, too.

**Comment by afkafkafk13** [comment]
Have you tried it with b before a in the list of variables?

**Comment by fingolfin** [comment]
Maybe @hannes14 can also have a look here?

What would be equivalent Singular code, and is it faster or also slow?

**Comment by fieker** [comment]
@jankoboehm will organise the plan:
 - a function in Oscar to test if singular can factor over a specific ring
 - go through the singular calls, use the function and error
additionally deal with the parallelism

**Comment by wdecker** [comment]
Interestingly enough, as @jankoboehm and I just tested, this uses (almost) no time:
```
R, (a, b, x, y, z) = QQ[:a, :b, :x, :y, :z]
g = QQMPolyRingElem[-x^3 + x*z^8 - x + y^2, a^8 + 1, -a^6 + a^4 - a^2 + b^2]
J = ideal(R, g)
minimal_primes(J, algorithm = :charSets)
```

We will further dig into this.

**Comment by wdecker** [comment]
@fieker There is no specific ring here!

**Comment by jankoboehm** [comment]
First note: This is not related to coefficient fields, but it seems that the GTZ which is the standard option for minimal_primes can not handle these type of ideals (created artifically to avoid multiple extensions when going to Singular?) not well. The Shimoyama-Yokoyama algorithm (minAssChar in Singular) can handle these type of ideals well. So one solution seems to be that whenever we create such rings to reduce to a coefficient ring which Singular can handle, we call this algorithmic option (assuming that the translation and the minimal_primes call are in the same place). Perhaps GTZ can be improved in future.

**Comment by wdecker** [comment]
@fieker Can you please check this example in Magma?

**Comment by fieker** [comment]
On Wed, Nov 13, 2024 at 04:19:20AM -0800, Wolfram Decker wrote:
> @fieker Can you please check this example in Magma?
Don't know...
Magma has
 PrimaryDecomposition(I) 
 RadicalDecomposition(I)
 MinimalDecomposition(S)

PD is fast here 0.110sec - but I don't the relation to minimal primes

QQa<a, b,x, y, z> := PolynomialRing(Rationals(), 5);
g := [-x^3 + x*z^8 - x + y^2, a^8 + 1, -a^6 + a^4 - a^2 + b^2];
J := Ideal(g);
time _ := PrimaryDecomposition(J);

**Comment by HereAround** [comment]
@hannes14 is working on it. Two observations:
1. Singular's approach is correct.
2. There are two algorithms in Singular, and the one that completes faster should - hopefully in the near future - be the one that returns the result.

This is WIP.

**Comment by wdecker** [comment]
We will also include a quick primality test which will apply in examples such as the one given by @HechtiDerLachs here.

**Comment by HereAround** [comment]
@hannes14 is still working on this. No progress yet. It appears that some theory considerations are needed, which @hannes14 will discuss with @wdecker and @jankoboehm .

**Comment by HereAround** [comment]
> We will also include a quick primality test which will apply in examples such as the one given by @HechtiDerLachs here.

We just discussed this in triage, and @wdecker voices that (upon closer inspection) it seems that such a quick primality test will not be feasible.

Other than that WIP and more soon.

**Comment by fingolfin** [comment]
@wdecker and @fieker will try a bit more to get an idea how to handle this, and possibly contact Allan Steel...

**Comment by fingolfin** [comment]
@wdecker @fieker so, did you contact Allan Steel or figure something out otherwise?
